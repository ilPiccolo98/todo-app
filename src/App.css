td,
th {
  border: black solid 1px;
}

.buttons-panel {
  margin-left: 160px;
}
