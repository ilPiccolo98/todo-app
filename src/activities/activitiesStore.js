import { configureStore } from "@reduxjs/toolkit";
import activitiesReducer from "./activitiesReducer";

const storeActivities = configureStore({
  reducer: {
    activities: activitiesReducer,
  },
});

export default storeActivities;
