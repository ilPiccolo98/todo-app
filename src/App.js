import { BrowserRouter, Link, Switch, Route } from "react-router-dom";
import TableActivities from "./TableActivities/TableActivities";
import AddActivity from "./AddActivity/AddActivity";
import "./App.css";
import { useState } from "react";
import UpdateActivity from "./UpdateActivity/UpdateActivity";
import DeleteActivity from "./DeleteActivity/DeleteActivity";

function App() {
  const [activities, setActivities] = useState([
    {
      id: 1,
      name: "shopping",
      description: "buy some stuff",
      status: false,
    },
    {
      id: 2,
      name: "have fun with my dog",
      description: "take my dog for a walk and have fun with it",
      status: true,
    },
    {
      id: 3,
      name: "work out",
      description: "work out a little bit with weights and barbells",
      status: false,
    },
  ]);
  return (
    <BrowserRouter>
      <TableActivities activities={activities} setActivities={setActivities} />
      <div className="buttons-panel">
        <Link to="/add-activity">
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </Link>
        <Link to="/update-activity">
          <input
            type="button"
            value="Update activity"
            data-testid="update-button-app"
          />
        </Link>
        <Link to="/delete-activity">
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </Link>
      </div>
      <Switch>
        <Route
          exact
          path="/add-activity"
          render={() => (
            <AddActivity
              activities={activities}
              setActivities={setActivities}
            />
          )}
        />
        <Route
          exact
          path="/update-activity"
          render={() => (
            <UpdateActivity
              activities={activities}
              setActivities={setActivities}
            />
          )}
        />
        <Route
          exact
          path="/delete-activity"
          render={() => (
            <DeleteActivity
              activities={activities}
              setActivities={setActivities}
            />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
