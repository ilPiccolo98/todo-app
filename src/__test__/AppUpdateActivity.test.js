import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import App from "../App";
import activitiesStore from "../activities/activitiesStore";

const renderAppActivity = () => {
  return render(
    <Provider store={activitiesStore}>
      <App />
    </Provider>
  );
};

const getUpdateActivityFormComponents = (component) => {
  const idFieldUpdateActivity = component.getByTestId(
    "id-field-update-activity"
  );
  const nameFieldUpdateActivity = component.getByTestId(
    "name-field-update-activity"
  );
  const descriptionFieldUpdateActivity = component.getByTestId(
    "description-field-update-activity"
  );
  const statusCheckboxUpdateActivity = component.getByTestId(
    "status-checkbox-update-activity"
  );
  const updateButtonUpdateActivity = component.getByTestId(
    "update-button-update-activity"
  );
  return {
    idFieldUpdateActivity,
    nameFieldUpdateActivity,
    descriptionFieldUpdateActivity,
    statusCheckboxUpdateActivity,
    updateButtonUpdateActivity,
  };
};

const getRowDataCells = (table, index) => {
  const idCellData = table.childNodes[index].childNodes[0];
  const nameCellData = table.childNodes[index].childNodes[1];
  const descriptionCellData = table.childNodes[index].childNodes[2];
  const statusCellData = table.childNodes[index].childNodes[3];
  return {
    idCellData,
    nameCellData,
    descriptionCellData,
    statusCellData,
  };
};

describe("testing App when updating a new activity", () => {
  it("shouldn't update an activity with the id field blank", () => {
    const component = renderAppActivity();
    const updateButtonApp = component.getByTestId("update-button-app");
    fireEvent.click(updateButtonApp);
    const {
      nameFieldUpdateActivity,
      descriptionFieldUpdateActivity,
      updateButtonUpdateActivity,
    } = getUpdateActivityFormComponents(component);
    fireEvent.change(nameFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.change(descriptionFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.click(updateButtonUpdateActivity);
    const table = component.getByTestId("table-body");
    for (let i = 0; i !== table.childElementCount; ++i) {
      const { nameCellData, descriptionCellData } = getRowDataCells(table, i);
      expect(nameCellData.textContent).not.toBe("activity updated");
      expect(descriptionCellData.textContent).not.toBe("activity updated");
    }
    expect(component).toMatchSnapshot();
  });

  it("shouldn't update an not existing activity", () => {
    const component = renderAppActivity();
    const updateButtonApp = component.getByTestId("update-button-app");
    fireEvent.click(updateButtonApp);
    const {
      idFieldUpdateActivity,
      nameFieldUpdateActivity,
      descriptionFieldUpdateActivity,
      statusCheckboxUpdateActivity,
      updateButtonUpdateActivity,
    } = getUpdateActivityFormComponents(component);
    fireEvent.change(idFieldUpdateActivity, {
      target: {
        value: "10",
      },
    });
    fireEvent.change(nameFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.change(descriptionFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.click(statusCheckboxUpdateActivity);
    fireEvent.click(updateButtonUpdateActivity);
    const table = component.getByTestId("table-body");
    for (let i = 0; i !== table.childElementCount; ++i) {
      const { nameCellData, descriptionCellData } = getRowDataCells(table, i);
      expect(nameCellData.textContent).not.toBe("activity updated");
      expect(descriptionCellData.textContent).not.toBe("activity updated");
    }
    expect(component).toMatchSnapshot();
  });

  it("should update the activity with id 2", () => {
    const component = renderAppActivity();
    const updateButtonApp = component.getByTestId("update-button-app");
    fireEvent.click(updateButtonApp);
    const {
      idFieldUpdateActivity,
      nameFieldUpdateActivity,
      descriptionFieldUpdateActivity,
      statusCheckboxUpdateActivity,
      updateButtonUpdateActivity,
    } = getUpdateActivityFormComponents(component);
    fireEvent.change(idFieldUpdateActivity, {
      target: {
        value: "2",
      },
    });
    fireEvent.change(nameFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.change(descriptionFieldUpdateActivity, {
      target: {
        value: "activity updated",
      },
    });
    fireEvent.click(statusCheckboxUpdateActivity);
    fireEvent.click(updateButtonUpdateActivity);
    const table = component.getByTestId("table-body");
    const { idCellData, nameCellData, descriptionCellData, statusCellData } =
      getRowDataCells(table, 2);
    expect(idCellData.textContent).toBe("2");
    expect(nameCellData.textContent).toBe("activity updated");
    expect(descriptionCellData.textContent).toBe("activity updated");
    expect(statusCellData.textContent).toBe("true");
    expect(component).toMatchSnapshot();
  });
});
