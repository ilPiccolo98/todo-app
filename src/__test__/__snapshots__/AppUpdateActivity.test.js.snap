// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App when updating a new activity should update the activity with id 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                activity updated
              </td>
              <td
                data-testid="td-description-2"
              >
                activity updated
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form>
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-update-activity"
            id="id"
            name="id"
            pattern="[0-9]*"
            placeholder="Id"
            type="text"
            value="2"
          />
        </div>
        <div>
          <label
            for="name"
          >
            Name
          </label>
          <input
            data-testid="name-field-update-activity"
            id="name"
            name="name"
            placeholder="Name"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description
          </label>
          <input
            data-testid="description-field-update-activity"
            id="description"
            name="description"
            placeholder="Description"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="status"
          >
            Status
          </label>
          <input
            data-testid="status-checkbox-update-activity"
            id="status"
            name="status"
            type="checkbox"
          />
        </div>
        <input
          data-testid="update-button-update-activity"
          type="submit"
          value="Update"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              activity updated
            </td>
            <td
              data-testid="td-description-2"
            >
              activity updated
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form>
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-update-activity"
          id="id"
          name="id"
          pattern="[0-9]*"
          placeholder="Id"
          type="text"
          value="2"
        />
      </div>
      <div>
        <label
          for="name"
        >
          Name
        </label>
        <input
          data-testid="name-field-update-activity"
          id="name"
          name="name"
          placeholder="Name"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="description"
        >
          Description
        </label>
        <input
          data-testid="description-field-update-activity"
          id="description"
          name="description"
          placeholder="Description"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="status"
        >
          Status
        </label>
        <input
          data-testid="status-checkbox-update-activity"
          id="status"
          name="status"
          type="checkbox"
        />
      </div>
      <input
        data-testid="update-button-update-activity"
        type="submit"
        value="Update"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing App when updating a new activity shouldn't update an activity with the id field blank 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form>
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-update-activity"
            id="id"
            name="id"
            pattern="[0-9]*"
            placeholder="Id"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="name"
          >
            Name
          </label>
          <input
            data-testid="name-field-update-activity"
            id="name"
            name="name"
            placeholder="Name"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description
          </label>
          <input
            data-testid="description-field-update-activity"
            id="description"
            name="description"
            placeholder="Description"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="status"
          >
            Status
          </label>
          <input
            data-testid="status-checkbox-update-activity"
            id="status"
            name="status"
            type="checkbox"
          />
        </div>
        <input
          data-testid="update-button-update-activity"
          type="submit"
          value="Update"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form>
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-update-activity"
          id="id"
          name="id"
          pattern="[0-9]*"
          placeholder="Id"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="name"
        >
          Name
        </label>
        <input
          data-testid="name-field-update-activity"
          id="name"
          name="name"
          placeholder="Name"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="description"
        >
          Description
        </label>
        <input
          data-testid="description-field-update-activity"
          id="description"
          name="description"
          placeholder="Description"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="status"
        >
          Status
        </label>
        <input
          data-testid="status-checkbox-update-activity"
          id="status"
          name="status"
          type="checkbox"
        />
      </div>
      <input
        data-testid="update-button-update-activity"
        type="submit"
        value="Update"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing App when updating a new activity shouldn't update an not existing activity 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table>
          <tbody
            data-testid="table-body"
          >
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
            <tr>
              <td
                data-testid="td-id-1"
              >
                1
              </td>
              <td
                data-testid="td-name-1"
              >
                shopping
              </td>
              <td
                data-testid="td-description-1"
              >
                buy some stuff
              </td>
              <td
                data-testid="td-status-1"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-2"
              >
                2
              </td>
              <td
                data-testid="td-name-2"
              >
                have fun with my dog
              </td>
              <td
                data-testid="td-description-2"
              >
                take my dog for a walk and have fun with it
              </td>
              <td
                data-testid="td-status-2"
              >
                true
              </td>
            </tr>
            <tr>
              <td
                data-testid="td-id-3"
              >
                3
              </td>
              <td
                data-testid="td-name-3"
              >
                work out
              </td>
              <td
                data-testid="td-description-3"
              >
                work out a little bit with weights and barbells
              </td>
              <td
                data-testid="td-status-3"
              >
                false
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttons-panel"
      >
        <a
          href="/add-activity"
        >
          <input
            data-testid="insert-button-app"
            type="button"
            value="Insert activity"
          />
        </a>
        <a
          href="/update-activity"
        >
          <input
            data-testid="update-button-app"
            type="button"
            value="Update activity"
          />
        </a>
        <a
          href="/delete-activity"
        >
          <input
            data-testid="delete-button-app"
            type="button"
            value="Delete activity"
          />
        </a>
      </div>
      <form>
        <div>
          <label
            for="id"
          >
            Id
          </label>
          <input
            data-testid="id-field-update-activity"
            id="id"
            name="id"
            pattern="[0-9]*"
            placeholder="Id"
            type="text"
            value="10"
          />
        </div>
        <div>
          <label
            for="name"
          >
            Name
          </label>
          <input
            data-testid="name-field-update-activity"
            id="name"
            name="name"
            placeholder="Name"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description
          </label>
          <input
            data-testid="description-field-update-activity"
            id="description"
            name="description"
            placeholder="Description"
            type="text"
            value="activity updated"
          />
        </div>
        <div>
          <label
            for="status"
          >
            Status
          </label>
          <input
            data-testid="status-checkbox-update-activity"
            id="status"
            name="status"
            type="checkbox"
          />
        </div>
        <input
          data-testid="update-button-update-activity"
          type="submit"
          value="Update"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <div>
      <table>
        <tbody
          data-testid="table-body"
        >
          <tr>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
          <tr>
            <td
              data-testid="td-id-1"
            >
              1
            </td>
            <td
              data-testid="td-name-1"
            >
              shopping
            </td>
            <td
              data-testid="td-description-1"
            >
              buy some stuff
            </td>
            <td
              data-testid="td-status-1"
            >
              false
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-2"
            >
              2
            </td>
            <td
              data-testid="td-name-2"
            >
              have fun with my dog
            </td>
            <td
              data-testid="td-description-2"
            >
              take my dog for a walk and have fun with it
            </td>
            <td
              data-testid="td-status-2"
            >
              true
            </td>
          </tr>
          <tr>
            <td
              data-testid="td-id-3"
            >
              3
            </td>
            <td
              data-testid="td-name-3"
            >
              work out
            </td>
            <td
              data-testid="td-description-3"
            >
              work out a little bit with weights and barbells
            </td>
            <td
              data-testid="td-status-3"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="buttons-panel"
    >
      <a
        href="/add-activity"
      >
        <input
          data-testid="insert-button-app"
          type="button"
          value="Insert activity"
        />
      </a>
      <a
        href="/update-activity"
      >
        <input
          data-testid="update-button-app"
          type="button"
          value="Update activity"
        />
      </a>
      <a
        href="/delete-activity"
      >
        <input
          data-testid="delete-button-app"
          type="button"
          value="Delete activity"
        />
      </a>
    </div>
    <form>
      <div>
        <label
          for="id"
        >
          Id
        </label>
        <input
          data-testid="id-field-update-activity"
          id="id"
          name="id"
          pattern="[0-9]*"
          placeholder="Id"
          type="text"
          value="10"
        />
      </div>
      <div>
        <label
          for="name"
        >
          Name
        </label>
        <input
          data-testid="name-field-update-activity"
          id="name"
          name="name"
          placeholder="Name"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="description"
        >
          Description
        </label>
        <input
          data-testid="description-field-update-activity"
          id="description"
          name="description"
          placeholder="Description"
          type="text"
          value="activity updated"
        />
      </div>
      <div>
        <label
          for="status"
        >
          Status
        </label>
        <input
          data-testid="status-checkbox-update-activity"
          id="status"
          name="status"
          type="checkbox"
        />
      </div>
      <input
        data-testid="update-button-update-activity"
        type="submit"
        value="Update"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
